tools/slang-cpp-extractor/cpp-extractor-main.cpp

```diff
@@ -14,14 +14,14 @@
 #include "../../source/core/slang-string-util.h"
 #include "../../source/core/slang-string.h"
 #include "../../source/core/slang-writer.h"
-#include "diagnostics.h"
-#include "file-util.h"
+#include "../tools/slang-cpp-parser/diagnostics.h"
+#include "../tools/slang-cpp-parser/file-util.h"
+#include "../tools/slang-cpp-parser/node.h"
+#include "../tools/slang-cpp-parser/options.h"
+#include "../tools/slang-cpp-parser/parser.h"
+#include "../tools/slang-cpp-parser/unit-test.h"
```

Comment:
Can we avoid using "../" here?
We already have a few but we should try to remove them not adding them.

Reply:
I think we can by passing `INCLUDE_DIRECTORIES_PUBLIC .` to the `slang-cpp-parser` target in the CMake file.
Should I remove the relative includes in all other modified files too? I just verified it's already possible to include directly from `core/`, `comiler-core/` and `slang/` in those. What annoys me with that though is the clang formatter intertwining includes from the target itself with includes from other targets by just sorting alphabetically.

Reply:
yeah, I think we ought to do exactly that and just `#include "slang-cpp-parser/node.h"` etc. If you want to change the other ones feel free.

I wouldn't worry about the include ordering, but if it really bothers you the I guess we could have a .clang-format file in this subdirectory which inherits from the root and specifies `IncludeCategories`?

Reply:
I am not sure how it can be done in cmake syntax but it sounds fine to me.

I am strongly against using the "../" for the include path. I have seen it including unexpected files with same file name before.

Reply:
I removed all relative includes.